### PR TITLE
PWN-3946 - UI Kit Tip

### PR DESCRIPTION
**Pull request by kuznetsov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-3946

## Description of Work

- Added `UiKitTip` utils.
- Tip could be shown under or above the anchor view.
- Arrow position is calculated programmatically to point the center of anchor view.

## Screenshots

Screenshot 2022-07-14 at 18 20 47 | Screenshot 2022-07-14 at 18 20 50 | Screenshot 2022-07-14 at 18 20 54
:------------:|:-------------:|:------------:

### CODE REVIEW

**Comment by kabramovp2p** [comment]
shadow for arrow on top is :poop:

**Comment by kuznetsov** [comment]
It looks pretty lame on non contrast background, but I think we will use it differently

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+}
+
+/**
+ * Shows PopupWindow with arrow
+ * PopupWindow appears natively above or under the view
+ * Arrow is shown only on top or bottom of PopupWindow
+ * Arrow is pointing on the center of [anchorView]
+ */
+fun showTip(
+    anchorView: View,
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt:27]: how's the idea make this as a extension to a achorView?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    counterText: String,
+    @StringRes title: Int,
+    @StringRes buttonNextTitle: Int = R.string.tip_button_next_title,
+    @StringRes buttonSkipTitle: Int = R.string.tip_button_skip_all_title,
+    onNextClick: () -> Unit,
+    onSkipClick: () -> Unit
+) {
+    val context = anchorView.context
+    showTip(
+        anchorView,
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt:37]: named arg?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        nextButton.setOnClickListener { onNextClick() }
+        skipAllButton.setOnClickListener { onSkipClick() }
+    }.apply {
+        root.measure(View.MeasureSpec.UNSPECIFIED, View.MeasureSpec.UNSPECIFIED)
+    }
+
+    val popupView = binding.root
+
+    val popupWindow = PopupWindow(popupView, popupView.measuredWidth, popupView.measuredHeight, true).apply {
+        elevation = 8f.toPx()
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt:76]: to const?

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        skipAllButton.text = buttonSkipTitle
+        nextButton.setOnClickListener { onNextClick() }
+        skipAllButton.setOnClickListener { onSkipClick() }
+    }.apply {
+        root.measure(View.MeasureSpec.UNSPECIFIED, View.MeasureSpec.UNSPECIFIED)
+    }
+
+    val popupView = binding.root
+
+    val popupWindow = PopupWindow(popupView, popupView.measuredWidth, popupView.measuredHeight, true).apply {
```
Comment (gslevinkov) [inline ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt:75]: i would do this constructor with separate line breaks for each param for Studio to show what each parameter means

**Review by gslevinkov — APPROVED** [review]

**Comment by gslevinkov** [comment]
no asignee

File: ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt (hunk 1 above)
Comment (kuznetsov) [reply ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt:27]: Sounds good. I'll do it

**Review by kabramovp2p — APPROVED** [review]

**Pull request merged**